File: server/repository/job.go
package repository

import (
	"context"
	"server/database"
	"server/graph/model"

	"github.com/google/uuid"
)

func CreateJob(ctx context.Context, input *model.CreateJob) (*model.Job, error) {
	db := database.GetDB()

	job := model.Job{
		ID:             uuid.NewString(),
		Title:          input.Title,
		CompanyName:    input.CompanyName,
		Workplace:      input.Workplace,
		Location:       &model.Location{
			Region: input.LocationRegion,
			City:   input.LocationCity,
		},
		EmploymentType: input.EmploymentType,
		Description:    input.Description,
	}

	if err := db.Create(&job).Error; err != nil {
		return nil, err
	}

	return &job, nil
}

File: client/src/graphql/job.ts
import { gql } from "@apollo/client";

export const JOBS = gql`
  query jobs {
    jobs {
      id
      title
      companyName
      workplace
      location {
        region
        city
      }
      employmentType
      description
    }
  }
`;

export const CREATE_JOB = gql`
  mutation createJob($input: CreateJob!) {
    job {
      create(input: $input) {
        id
      }
    }
  }
`;
